src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -7258,19 +7259,45 @@ subroutine summarize_timestep(domain)
        integer, dimension(:), pointer :: indexToCellID
        integer :: indexMax, indexMax_global
        integer :: kMax, kMax_global
+       integer :: offset_1d ! Offset into a multi-dimensional array, as if it were a contiguous 1-d array
        real (kind=RKIND) :: latMax, latMax_global
        real (kind=RKIND) :: lonMax, lonMax_global
        real (kind=RKIND), dimension(5) :: localVals, globalVals
 
-       real (kind=RKIND) :: spd
+       real (kind=RKIND), dimension(:,:), allocatable :: spd
+       !$acc declare create(spd)
        real (kind=RKIND), dimension(:,:), pointer :: w
        real (kind=RKIND), dimension(:,:), pointer :: u, v, uReconstructZonal, uReconstructMeridional, uReconstructX, uReconstructY, uReconstructZ
        real (kind=RKIND), dimension(:,:,:), pointer :: scalars, scalars_1, scalars_2
 
+       logical :: found_NaN
+
        call mpas_pool_get_config(block % configs, 'config_print_global_minmax_vel', config_print_global_minmax_vel)
        call mpas_pool_get_config(block % configs, 'config_print_detailed_minmax_vel', config_print_detailed_minmax_vel)
        call mpas_pool_get_config(block % configs, 'config_print_global_minmax_sca', config_print_global_minmax_sca)
 
+      ! Fetch pointers early for temporary OpenACC data transfers
```

Comment:
I think this comment is unnecessary, and at a minimum, I'd suggest removing the words "early" and "temporary".

Reply:
I assumed these `..._get_array` calls were as temporary as the `acc (enter|exit) data` directives they enable, so I just duplicated them to keep the code modifications to a minimum.

I'll get the comment and the later `...get_array` calls removed.

Reply:
I think you make a fair point that we're only moving these `mpas_pool_get_array` calls so that we can place the `!$acc enter data` directives at the beginning of the subroutine, and our goal is to eventually eliminate those data movement directives. On the one hand, we could move the `mpas_pool_get_array` calls back to where they originally were (either now, or after eliminating data movement directives); or, we could just leave them here at the beginning of the routine, which might be ultimately be cleaner.

Reply:
I'm all here for cleaner code! Given our talk earlier this week about there being nearly 0 impact from `mpas_pool_get_array` calls, I don't see a problem with occasionally fetching more than we need.

I don't think there would be any impacts on the GPU execution to fetch the (host-side) pointers if we don't use them. There should only be data movement once something causes an attach action - a compute region (esp. one with `default(present)`, a `copyin`, a `present` clause on a data region, etc.